examples/z_sub.c

```diff
@@ -22,10 +23,19 @@
 
 const char *kind_to_str(z_sample_kind_t kind);
 
+int8_t attachement_handler(z_bytes_t key, z_bytes_t value, void *ctx) {
+    (void)ctx;
+    printf(">> [Subscriber]   attached %.*s -> %.*s", (int)key.len, key.start, (int)value.len, value.start);
+    return 0;
+}
+
 void data_handler(const z_sample_t *sample, void *arg) {
     z_owned_str_t keystr = z_keyexpr_to_string(sample->keyexpr);
     printf(">> [Subscriber] Received %s ('%s': '%.*s')\n", kind_to_str(sample->kind), z_loan(keystr),
            (int)sample->payload.len, sample->payload.start);
+    if (z_check(sample->attachements)) {
```

Comment:
In the `put` example I propose to have `z_attachment_insert(..)`. In the sub, what about something like:

```c
z_attachment_value v = z_attachment_get(sample->attachment, z_attachment_key("hello"));
if (z_check(v)) {
  // Do your stuff
}
```

We might still implement the internals of a `z_attachment_t` as a `z_bytes_map`, but that would be abstracted  by the `z_attachment_key` and `z_attachment_value`. By doing so, we reserve a bit of freedom to change the internal implementation without changing the API.

Reply:
The goal here is for `z_attachement_t` to be a generic, read-only type. This lets users pick whatever suits them best as a concrete implementation. The map is provided as the default way to make up an attachment, but that way of working lets us have lazy deserializers on the read side, and if the user want noalloc attachements, they can have them